Tools/Create-Azure-Sentinel-Solution/common/createCCPConnector.ps1

```diff
@@ -0,0 +1,880 @@
+                            if ($placeHoldersMatched.Matches.Value.Count -gt 0) {
+                                $placeHolderName = $placeHoldersMatched.Matches.Value.replace("{{", "").replace("}}", "")
+                                $armResource.properties.request.AuthorizationEndpoint = "[[parameters('AuthorizationEndpoint')]"
+                                $templateContentConnections.properties.mainTemplate = addNewParameter -templateResourceObj $templateContentConnections.properties.mainTemplate -parameterName 'AuthorizationEndpoint' -isSecret $false
+                            }
+                        }
+    
+                        # TokenEndpoint placeholder 
+                        if ($null -ne $armResource.properties.auth.TokenEndpoint -and $armResource.properties.auth.TokenEndpoint.contains("{{")) {
+                            $tokenEndpointValue = $armResource.properties.auth.TokenEndpoint
+                            $placeHoldersMatched = $tokenEndpointValue | Select-String $placeHolderPatternMatches -AllMatches
+                            if ($placeHoldersMatched.Matches.Value.Count -gt 0) {
+                                $placeHolderName = $placeHoldersMatched.Matches.Value.replace("{{", "").replace("}}", "")
+                                $armResource.properties.auth.TokenEndpoint = "[[parameters('TokenEndpoint')]"
+                                $templateContentConnections.properties.mainTemplate = addNewParameter -templateResourceObj $templateContentConnections.properties.mainTemplate -parameterName 'TokenEndpoint' -isSecret $false
+                            }
+                        }
+                    }
+                    elseif ($armResource.properties.auth.type.ToLower() -eq 'basic') {
+                        # username
+                        $hasUsername = [bool]($armResource.properties.auth.PSobject.Properties.name -match "username")
+                        if ($hasUsername) {
+                            $usernameProperty = $armResource.properties.auth.username
+                            $placeHoldersMatched = $usernameProperty | Select-String $placeHolderPatternMatches -AllMatches
+    
+                            if ($placeHoldersMatched.Matches.Value.Count -gt 0) {
+                                $placeHolderName = $placeHoldersMatched.Matches.Value.replace("{{", "").replace("}}", "")
+                                $armResource.properties.auth.username = "[[parameters('username')]"
+    
+                                $templateContentConnections.properties.mainTemplate = addNewParameter -templateResourceObj $templateContentConnections.properties.mainTemplate -parameterName 'username' -isSecret $false
+                            }
+                        }
+    
+                        # password 
+                        $hasPassword = [bool]($armResource.properties.auth.PSobject.Properties.name -match "password")
+                        if ($hasPassword) {
```

Comment:
For normal string parameters, which has type password this code has to be reused to define type as secure string

Reply:
updated